pkg/policy/l4_test.go

```diff
@@ -143,6 +143,9 @@ func (s *PolicyTestSuite) TestJSONMarshal(c *C) {
 	expectedIngress := []string{`{
   "port": 80,
   "protocol": "TCP",
+  "l3-selectors": [
+    "\u0026LabelSelector{MatchLabels:map[string]string{any.foo: ,},MatchExpressions:[],}"
```

Comment:
IMO, I don't like exposing this string representation of the endpoint selector in the endpoint API, there is a lot of boilerplate that makes this unnecessarily hard to read. Maybe we should make `String` method of a `CachedSelector` print out something prettier? This is likely out of scope for this PR, though.

Reply:
Apparently the l7-rules are already rendering like this, and yeah it should probably be fixed there too but probably doesn't need to be tied to this PR

Maybe fixing this isn't too difficult - I presume this is coming out from the `buf, err := json.Marshal(selector.String())` line in `selectorcache.go`, can we either render each selector into json in a buffer first then stringify this, or maybe there's already some more friendly stringification function in the underlying upstream k8s selectors?